src/telemetry/metrics/mod.rs

```diff
@@ -209,58 +211,58 @@ impl Root {
         }
     }
 
-    fn request(&mut self, labels: RequestLabels) -> &mut http::RequestMetrics {
-        self.requests.scopes.entry(labels)
+    fn request<F: FnOnce(&mut http::RequestMetrics)>(&self, labels: RequestLabels, f: F) {
+        f(self.requests.lock().expect("lock").scopes.entry(labels)
```

Comment:
Rather than taking a closure, I think these functions could alternatively just return a `MutexGuard<WhateverMetrics>`? I'm not sure if this would be significantly better or not.